autometa/taxonomy/ncbi.py

```diff
@@ -399,24 +406,26 @@ def main():
     parser.add_argument('ncbi', help='</path/to/ncbi/database/directory>')
     parser.add_argument(
         '--query-taxid',
-        help='query taxid to test NCBI class functionality',
+        help='query taxid to retrieve its name, order, class, rank and complete lineage',
         default=1222,
-        type=int,
+        type=int
     )
     parser.add_argument('--verbose', help="add verbosity",
-                        action='store_true', default=True)
+                        action='store_true', default=False)
     args = parser.parse_args()
     ncbi = NCBI(args.ncbi, args.verbose)
     query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f'{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f'{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} is_common_ancestor {query_taxid_parent}: '
-        f'{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n'
-    )
+    # Done to do a plimenary check if taxid is valid or not. Prevents recurvise printing of error statements
+    if query_taxid_parent:
+        logger.info(
+            f'{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n'
+            f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
+            f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
+            f'{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n'
+            f'{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n'
+            f'{args.query_taxid} is_common_ancestor {query_taxid_parent}: '
+            f'{ncbi.is_common_ancestor(taxid1=query_taxid_parent, taxid2=args.query_taxid)}\n'
+        )
```

Comment:
I think you can entirely remove `argparse` from this similar to `utillities.py` if we do not want to run this as a standalone module.